### PR TITLE
Rework first-class AWS login support

### PR DESCRIPTION
**Pull request by benashz**

### CODE REVIEW

File: internal/consts/consts.go [hunk 1]
```diff
@@ -79,14 +79,22 @@ const (
...
 	FieldAuthLoginAzure           = "auth_login_azure"
-	FieldIdentity                 = "identity"
-	FieldSignature                = "signature"
-	FieldPKCS7                    = "pkcs7"
-	FieldNonce                    = "nonce"
 	FieldIAMHttpRequestMethod     = "iam_http_request_method"
 	FieldIAMHttpRequestURL        = "iam_http_request_url"
 	FieldIAMHttpRequestBody       = "iam_http_request_body"
 	FieldIAMHttpRequestHeaders    = "iam_http_request_headers"
+	FieldAWSAccessKeyID           = "aws_access_key_id"
```
Comment (vinay-gopalan) [inline internal/consts/consts.go:86]: nit: I wonder if we should now rename the pre-existing constants defined here to be more generic. Like `FieldAccessKey` and `FieldSecretKey` https://github.com/hashicorp/terraform-provider-vault/blob/main/internal/consts/consts.go#L48

**Review by vinay-gopalan — APPROVED** [review]
LGTM

File: internal/consts/consts.go (hunk 1 above)
Comment (benashz) [reply internal/consts/consts.go:86]: Funny you should say that, I did just that for the Login MFA work.

Comment (vinay-gopalan) [reply internal/consts/consts.go:86]: Nice!

**Pull request merged**